src/main/kotlin/com/terraformation/backend/support/atlassian/resource/CreateServiceDeskRequest.kt

```diff
@@ -0,0 +1,116 @@
+package com.terraformation.backend.support.atlassian.resource
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.support.atlassian.SupportRequestType
+import io.ktor.client.call.body
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.HttpResponse
+import io.ktor.http.HttpMethod
+import io.ktor.http.path
+import javax.inject.Named
+
+/**
+ * The POST request to create a service ticket.
+ *
+ * @param requestTypeId the ID of the JIRA issue type
+ * @param summary the title of the JIRA issue
+ * @param description the details of the JIRA issue
+ * @param reporter the email address of the reporter
+ */
+data class CreateServiceDeskRequest(
+    val description: String,
+    val summary: String,
+    val reporter: String? = null,
+    val requestTypeId: Int,
+    val serviceDeskId: Int
+) : AtlassianResource<PostServiceDeskRequestResponse> {
+  private val path = "/rest/servicedeskapi/request"
+  private val httpMethod = HttpMethod.Post
+
+  override fun buildRequest(requestBuilder: HttpRequestBuilder) {
+    val body =
+        PostServiceDeskRequestBody(
+            description = description,
+            requestTypeId = requestTypeId,
+            summary = summary,
+            raiseOnBehalfOf = reporter,
+            serviceDeskId = serviceDeskId,
+        )
+    with(requestBuilder) {
+      method = httpMethod
+      url { path(path) }
+      setBody(body)
+    }
+  }
+
+  override suspend fun parseResponse(response: HttpResponse): PostServiceDeskRequestResponse {
+    return response.body()
+  }
+
+  /** A builder class to inject configs or build information into the resource. */
+  @Named
+  class Builder(private val config: TerrawareServerConfig) {
```

Comment:
Can we get away with not having a builder class at all?

In Java we needed this kind of thing thanks to the lack of named parameters, but this could be a secondary constructor on the outer class to do the request type mapping, plus an `init` block to do the validation checks.

The big advantage is you get compile-time checking of required arguments. WIth a builder, forgetting to set one of the mandatory values is a runtime error (the `!!` will throw an exception) rather than a compile failure.

Reply:
I agree. I was struggling a bit to inject TerrawareServer config (and maybe sometime in the future things like BuildVersion) into the request constructor. I do think Builder is not the correct pattern. I am leaning towards just a RequestFactory class. Will follow up in the next revision and get your feedback on that.